repos/system_upgrade/common/libraries/rhui.py

```diff
@@ -18,10 +21,234 @@
+        mk_rhui_setup(clients={'rh-amazon-rhui-client'}, leapp_pkg='leapp-rhui-aws',
+                      mandatory_files=[
+                        ('rhui-client-config-server-9.crt', RHUI_PKI_PRODUCT_DIR),
+                        ('rhui-client-config-server-9.key', RHUI_PKI_DIR),
+                        ('leapp-aws.repo', YUM_REPOS_PATH)
+                      ],
+                      optional_files=[], os_version='9'),
+    ],
+    RHUIFamily(RHUIProvider.AWS, arch=arch.ARCH_ARM64, client_files_folder='aws'): [
+        mk_rhui_setup(clients={'rh-amazon-rhui-client-arm'}, optional_files=[], os_version='7', arch=arch.ARCH_ARM64),
+        mk_rhui_setup(clients={'rh-amazon-rhui-client-arm'}, leapp_pkg='leapp-rhui-aws',
+                      mandatory_files=[
+                         ('rhui-client-config-server-8.crt', RHUI_PKI_PRODUCT_DIR),
+                         ('rhui-client-config-server-8.key', RHUI_PKI_DIR),
+                         (AWS_DNF_PLUGIN_NAME, DNF_PLUGIN_PATH_PY2),
+                         ('leapp-aws.repo', YUM_REPOS_PATH)
+                      ],
+                      files_supporting_client_operation=[AWS_DNF_PLUGIN_NAME],
+                      optional_files=[], os_version='8', arch=arch.ARCH_ARM64),
+        mk_rhui_setup(clients={'rh-amazon-rhui-client-arm'}, leapp_pkg='leapp-rhui-aws',
+                      mandatory_files=[
+                        ('rhui-client-config-server-9.crt', RHUI_PKI_PRODUCT_DIR),
+                        ('rhui-client-config-server-9.key', RHUI_PKI_DIR),
+                        ('leapp-aws.repo', YUM_REPOS_PATH)
+                      ],
+                      optional_files=[], os_version='9', arch=arch.ARCH_ARM64),
+    ],
+    RHUIFamily(RHUIProvider.AWS, variant=RHUIVariant.SAP, client_files_folder='aws-sap-e4s'): [
+        mk_rhui_setup(clients={'rh-amazon-rhui-client-sap-bundle'}, optional_files=[], os_version='7',
+                      content_channel=ContentChannel.E4S),
+        mk_rhui_setup(clients={'rh-amazon-rhui-client-sap-bundle-e4s'}, leapp_pkg='leapp-rhui-aws-sap-e4s',
+                      mandatory_files=[
+                        ('rhui-client-config-server-8-sap-bundle.crt', RHUI_PKI_PRODUCT_DIR),
+                        ('rhui-client-config-server-8-sap-bundle.key', RHUI_PKI_DIR),
+                        (AWS_DNF_PLUGIN_NAME, DNF_PLUGIN_PATH_PY2),
+                        ('leapp-aws-sap-e4s.repo', YUM_REPOS_PATH)
+                      ],
+                      optional_files=[], os_version='8', content_channel=ContentChannel.E4S),
+        mk_rhui_setup(clients={'rh-amazon-rhui-client-sap-bundle-e4s'}, leapp_pkg='leapp-rhui-aws-sap-e4s',
+                      mandatory_files=[
+                        ('rhui-client-config-server-9-sap-bundle.crt', RHUI_PKI_PRODUCT_DIR),
+                        ('rhui-client-config-server-9-sap-bundle.key', RHUI_PKI_DIR),
+                        ('cdn.redhat.com-chain.crt', RHUI_PKI_DIR),
```

Comment:
This is no longer a mandatory file I believe.

Reply:
@MichalHe +1 to @PeterMocary , missing cdn.redhat.com-chain.crt fails for 8.6to9.0 and 8.8to9.2 on aws, removing that line allows to proceed and complete the upgrade successfully. 
For 7to8 on aws case, it fails with a missing gpg key and missing target repo errors, vms have been shared. Please see your slack PM for details.